libparsec/crates/client/src/certif/list.rs

```diff
@@ -151,6 +131,90 @@ pub(super) async fn list_users(
         .await?
 }
 
+#[derive(Debug, thiserror::Error)]
+pub enum CertifGetUserInfoError {
+    #[error("No user with this UserID")]
+    NonExisting,
+    #[error("Component has stopped")]
+    Stopped,
+    #[error(transparent)]
+    Internal(#[from] anyhow::Error),
+}
+
+impl From<CertifStoreError> for CertifGetUserInfoError {
+    fn from(value: CertifStoreError) -> Self {
+        match value {
+            CertifStoreError::Stopped => Self::Stopped,
+            CertifStoreError::Internal(err) => err.into(),
+        }
+    }
+}
+
+pub(super) async fn get_user_info_for_user_certificates(
+    user_certificate: std::sync::Arc<UserCertificate>,
+    revoked_certificate: Option<std::sync::Arc<RevokedUserCertificate>>,
+    update_certificate: Option<std::sync::Arc<UserUpdateCertificate>>,
+) -> UserInfo {
+    let (revoked_on, revoked_by) = match revoked_certificate {
+        None => (None, None),
+        Some(certif) => (Some(certif.timestamp), Some(certif.author.to_owned())),
+    };
+    let current_profile = match update_certificate {
+        None => user_certificate.profile,
+        Some(certif) => certif.new_profile,
```

Comment:
```suggestion
        Some(update_certificate) => update_certificate.new_profile,
```
(I'd rather make it extra clear the split between `user_certificate` vs `update_certificate` on where the profile is fetched)